source/PlayerInfo.cpp

```diff
@@ -1841,7 +1845,11 @@ void PlayerInfo::Visit(const System *system)
 	
 	visitedSystems.insert(system);
 	seen.insert(system);
-	double jumpRange = (flagship && flagship->JumpRange()) ? flagship->JumpRange() : System::DEFAULT_NEIGHBOR_DISTANCE;
+	double jumpRange;
+	if(flagship)
+		jumpRange = flagship->JumpRange();
+	if(!jumpRange)
+		jumpRange = System::DEFAULT_NEIGHBOR_DISTANCE;
```

Comment:
This code looks similar as the code above in AddChanges(changedSystems==true), can't we just call Visit again in the code above to avoid code duplication?

Reply:
I don't see how Visit could be called from AddChanges. Doing so would just keep recalculating the player's jump range (which is what I wanted to avoid in one of the latest commits). If we instead hand the jump range off to Visit from AddChanges, then wherever Visit gets called elsewhere then needs to calculate the jump range, so the code will just be duplicated elsewhere.

Reply:
Okay, this is a trade-off between performance and some code-duplication. The systems-changed/Visit code is not highly performance critical, but I'm fine with you making this choice here.